### ISSUE TITLE
How to reproduce the results when using Conv4 and neg-cosine?

### ISSUE THREAD

**Issue by wang22ti-666** [bl0/negative-margin.few-shot#9]
Hi, I am trying to reproduce the results when using Conv4 as the backbone and neg-cosine as the metric. However, the results seem to be lower than the reported results on miniImagenet:

> --------------------------------------------------------------------------------
> | train acc | 1 shot val | 5 shot val | 1 shot test | 5 shot test |
> | 55.79% | 47.05% ± 0.75% | 65.36% ± 0.69% | 46.96% ± 0.76% | 64.77% ± 0.65% |
> --------------------------------------------------------------------------------

and the parameters are set as:
```
{'dataset': {'base_file': 'data/miniImagenet/base.json',
             'base_val_file': '',
             'novel_file': 'data/miniImagenet/novel.json',
             'num_class': 64,
             'type': 'miniImagenet',
             'val_file': 'data/miniImagenet/val.json'},
 'method': {'backbone': 'Conv4',
            'image_size': 84,
            'metric': 'cosineface',
            'metric_params': CfgNode({'scale_factor': 30.0, 'margin': -0.02}),
            'metric_params_test': CfgNode({'scale_factor': 5, 'margin': 0.0})},
 'misc': {'checkpoint_dir': './output/miniImagenet/exp_neg-cosinemargin-0.02/checkpoints',
          'evaluate': False,
          'log_dir': './output/miniImagenet/exp_neg-cosinemargin-0.02/tensorboard',
          'num_workers': 8,
          'output_dir': './output/miniImagenet/exp_neg-cosinemargin-0.02',
          'resume': '',
          'rng_seed': 0},
 'test': {'batch_size': 1024,
          'finetune_params': {'iter': 100,
                              'optim': 'SGD',
                              'sgd_params': {'dampening': 0.9,
                                             'lr': 0.01,
                                             'momentum': 0.9,
                                             'weight_decay': 0.001}},
          'n_query': 16,
          'n_support': [1, 5],
          'n_way': 5,
          'num_episode': 600,
          'split': 'novel'},
 'train': {'adam_params': CfgNode({'lr': 0.003, 'weight_decay': 0.0}),
           'batch_size': 256,
           'drop_rate': 0.0,
           'dropblock_size': 5,
           'lr_scheduler': 'warmup_cosine',
           'optim': 'Adam',
           'print_freq': 10,
           'sgd_params': {'lr': 1.0,
                          'momentum': 0.9,
                          'nesterov': True,
                          'weight_decay': 0.0005},
           'stop_epoch': 400,
           'warmup_params': CfgNode({'multiplier': 16, 'epoch': 100})},
 'val': CfgNode({'num_episode': 100, 'freq': 20, 'n_support': 5})}
```
I get similar results when changing the margin to -0.2. 
Are there some problems with the above parameters?
Appreciate your reply, thanks!

**Comment by bl0** [comment]
Hi, sorry for the inconvenience. For Conv4, the optimal training parameter we use in the paper is
```
scale factor = 20.0
margin = -0.05 or -0.1
adam weight decay = 1e-5
```
Here are the raw results we got using above configs:
| margin | Train acc | Val acc | 1 shot val      | 5 shot val      | 1 shot test     | 5 shot test      |
| ------ | --------- | ------- | --------------- | --------------- | --------------- | --------------- |
| -0.05  | 63.49%    | 66.01%  | 52.52% +- 0.81% | 71.40% +- 0.63% | 52.84% +- 0.76% | 69.88% +- 0.67% |
| -0.1   | 63.26%    | 65.46%  | 52.94% +- 0.84% | 71.58% +- 0.68% | 52.38% +- 0.75% | 70.41% +- 0.66% |

**Comment by bl0** [comment]
I am closing this issue due to a long time of inactivity. Feel free to reopen it if there is any question. Thanks!
